### PR TITLE
Rename, Update Docs and Prepare for Release

### PR DESCRIPTION
**Pull request by XPhyro**
Renamed sxiv to nsxiv in the codebase. See #1 for previous discussions.

### CODE REVIEW

**Comment by GRFreire** [comment]
Nicely done. I've just added the binary to the `.gitignore`.

One thing to be aware is the readme is still target to the old sxiv - both links and description. Do you think this should be done in this PR or in other?

**Comment by XPhyro** [comment]
> One thing to be aware is the readme is still target to the old sxiv - both links and description. Do you think this should be done in this PR or in other?

Good catch, I thought I already did. I'll update the PR shortly.

**Comment by GRFreire** [comment]
Forgot to mention, but we might want to change the `CHANGELOG.md` file as well. There are links pointing to the [muennich repo](https://github.com/muennich/sxiv)

**Comment by XPhyro** [comment]
> Forgot to mention, but we might want to change the `CHANGELOG.md` file as well. There are links pointing to the [muennich repo](https://github.com/muennich/sxiv)

That is intentional. See [this](https://github.com/nsxiv/nsxiv/pull/1#discussion_r705983518) discussion in #1.

**Comment by GRFreire** [comment]
> > Forgot to mention, but we might want to change the `CHANGELOG.md` file as well. There are links pointing to the [muennich repo](https://github.com/muennich/sxiv)
> 
> That is intentional. See [this](https://github.com/nsxiv/nsxiv/pull/1#discussion_r705983518) discussion in #1.

This discussion have moved forward a bit, looks like its ok to replace the links from muennich to nsxiv

**Comment by N-R-K** [comment]
For the sake of unifying the discussion, I'll reiterate what I've said on the other PR, I'm fine with replacing the links in the changelog. However the text should NOT be changed.

- - -

On a separate note, let's roll this PR out once all the smaller patches are in. Some of the bigger patches (like animated webp) will need massive refactoring.

**Comment by XPhyro** [comment]
Situation update: We have one PR left for the rename to be rolled, #19.

**Comment by bakkeby** [comment]
This and #1 seems to be mostly about just replacing sxiv with nsxiv.

That is OK for now, but we should have a review of the general info and refine it for this project, e.g. this isn't really the case anymore:
> The sole purpose of nsxiv is to be the perfect image viewer for me.

As for the changelog I think this should be blank until we publish our first release. For sxiv we should refer to the previous (original) project.

**Comment by N-R-K** [comment]
This PR already touches some more things than just rename, e.g updating maintainers in manpage. So we can just rename this to `Rename and update docs` instead and do other doc related changes here too. I'm fine with that.

As for changelog, if no one opposes, then we should move it to a new file `CHANGELOG` and have the format be like this:

```
Nsxiv changelog: 
/* our change log goes here */
...
...
...


Sxiv changelog:
/* original change log goes here */
...
...
```

**Comment by XPhyro** [comment]
The bump commit is tagged with `v27`, I am assuming it's not shown because it's a PR and that it will be tagged correctly when merged.

File: CHANGELOG.md
Comment (N-R-K) [inline CHANGELOG.md:20]:
```suggestion
Changes:
  * Re-release under the name nsxiv
  * Sxiv.foreground and Sxiv.background changed to Nsxiv.win.foreground and Nsxiv.win.background
  * Rework the build system (#19)
  * Update documentation
Added:
  * Implement fill scale mode (#2)
  * Set `_NET_WM_PID` and `WM_CLIENT_MACHINE` X properties (#13)
  * Support custom bar colors (#19)
  * Toggle animation playback with Ctrl-a (#33)
Fixes:
  * Prevent crash when embedded into transparent window (#3)
  * Set gamma with `-G` (#31)
  * Rare crash when showing some GIFs (#41)
```

File: README.md
Comment (N-R-K) [inline README.md:8]: We need to change this.

**Review by N-R-K — CHANGES_REQUESTED** [review]

**Comment by N-R-K** [comment]
> The bump commit is tagged with `v27`, I am assuming it's not shown because it's a PR and that it will be tagged correctly when merged.

`git push --tags`

**Comment by N-R-K** [comment]
Also don't tag now. We'll tag after it's been squashed @XPhyro 

And can you create a `CONTRIBUTING.md` too? following this, since no one opposed https://github.com/nsxiv/nsxiv/discussions/39

And mention it on the README as well.

**Comment by XPhyro** [comment]
> `git push --tags`

It does show the tag on my fork, just not on the PR.

> Also don't tag now. We'll tag after it's been squashed @XPhyro

Sure.

> And can you create a `CONTRIBUTING.md` too? following this, since no one opposed #39
> And mention it on the README as well.

On it.

**Comment by XPhyro** [comment]
Converted TODO to markdown (TODO.md) for consistency with the other files.

File: README.md
Comment (N-R-K) [inline README.md:8]: I think this can be worded a bit better. Although I don't have any suggestions atm.

File: TODO.md
Comment (N-R-K) [inline TODO.md:8]: If we're going markdown, might as well use the fancy todo list feature.

```suggestion
- [ ] Load all frames from TIFF files. We have to write our own loader for this to
    happen--just like we did for GIF images--because Imlib2 does not support
    multiple frames. Issue #241.
- [ ] Add support for more embedded thumbnail formats. Right now, nsxiv seems to use
    the smallest one. Issue #238.
```

File: CONTRIBUTING.md
Comment (N-R-K) [inline CONTRIBUTING.md:23]:
```suggestion
See [TODO.md](TODO.md) or the [open issues](https://github.com/nsxiv/nsxiv/issues) to find something to work on.
```

**Review by N-R-K — CHANGES_REQUESTED** [review]

File: TODO.md
Comment (N-R-K) [reply TODO.md:8]: Ahh, also realized. `Issue #241.`,  `Issue #241` these won't work. We'll need to hardlink to sxiv's repo.

Comment (XPhyro) [reply TODO.md:8]: Wouldn't we remove the entry after it's done? I don't think `- [ ]` is necessary here.

File: README.md
Comment (N-R-K) [inline README.md:8]:
```suggestion
nsxiv is a fork of now unmaintained [sxiv](https://github.com/muennich/sxiv)
with the purpose of maintaining it and adding simple, sensible features.
nsxiv is free software licensed under GPLv2 and aims to be easy to modify and customize.
```

**Review by N-R-K — APPROVED** [review]

**Comment by N-R-K** [comment]
If we squash this, will it contain the `v27` tag?

**Comment by XPhyro** [comment]
> If we squash this, will it contain the `v27` tag?

Based on [this answer](https://stackoverflow.com/a/12279290/8403070), no. Apparently GitHub does not track tags in PRs.

**Comment by N-R-K** [comment]
@XPhyro https://github.com/XPhyro/nsxiv/tree/rename seems like we're still using the old sxiv logo on the README ?

**Comment by XPhyro** [comment]
> @XPhyro https://github.com/XPhyro/nsxiv/tree/rename seems like we're still using the old sxiv logo on the README ?

We are. Missed that one.

**Comment by XPhyro** [comment]
Can't update it here since it's on a different branch (gh-pages), see #45.

**Comment by N-R-K** [comment]
While we're here, do we also keep the current previews? Since we've made some visual changes (bar colors to be specific) it might be a good idea to show it off.

**Comment by XPhyro** [comment]
> While we're here, do we also keep the current previews? Since we've made some visual changes (bar colors to be specific) it might be a good idea to show it off.

@N-R-K Should I use the default colours, or change them? The default is white background, white bar and black text.

**Comment by N-R-K** [comment]
> I think I should change it and have them different between the image and thumbnail previews, such that it is obvious that it can be changed.

Yeah sounds good.

Here's one from me, it's low res tho. But it shows

* custom bar colors
* new X window title

If someone can get a higher resolution screenshot that does the same, that would be nice.


image

**Comment by N-R-K** [comment]
Other than new previews and moving to `src/` and `obj/` is there anything else we need to change before release?

**Comment by GRFreire** [comment]
The icons are still the old (sxiv) ones. I think that some were trying to solve this editing `data.h`, but had no success yet.

**Comment by XPhyro** [comment]
How's this?

**Comment by N-R-K** [comment]
> How's this?

A bit too high res. can you make the nsxiv bar and your wm bar bigger so that it can be read when embedded into the readme?

**Comment by N-R-K** [comment]
Also let's keep the [old first screenshot](https://github.com/muennich/sxiv/blob/gh-pages/img/image.png) in there and add our new one below it replacing the old thumb one.

**Comment by XPhyro** [comment]
I noticed now that we did not change `Nsxiv.font`. That should probably be `Nsxiv.bar.font`.

**Comment by GRFreire** [comment]
I was thinking about something here. Currently all images are stored in the `gh-pages` branch, but it is not working the best way in my opinion:
- We could just store all images in the same branch, in a `README` folder for example.
    - The problem to this is that whenever someone wants to get the source code for compiling, it will also download all images (not a problem for me but probably was for muennich when he did this).
- We could actually use `gh-pages` the way it should work, by hosting the images (maybe the readme and man page, but that's for another topic). The way this pr is loading the images is by using the link in the `raw` subdomain from github (`https://raw.githubusercontent.com/nsxiv/nsxiv/gh-pages/img/image.png `), not actually by the `gh-pages` website server. It just not look right to me.

What do you guys think about this? Am I making my self clear here on the problem or is this confusing?

**Comment by XPhyro** [comment]
I think it's nice to have these README-only images in a separate branch. Though we could use GH pages as intended too.

**Comment by N-R-K** [comment]
> I noticed now that we did not change `Nsxiv.font`. That should probably be `Nsxiv.bar.font`

Is the bar the only place where font is used? If so then I can quickly raise a PR.

- - -

@GRFreire I can see why you have an issue with this, we even forgot to update the icon because it was on a different branch. But imo it's not a good idea to put the images into the master branch. It will get included into the release tarball bloating up the download size for no reason.

**Comment by GRFreire** [comment]
> But imo it's not a good idea to put the images into the master branch

> I think it's nice to have these README-only images in a separate branch

I see, so let's spin up the `gh-pages` "_server_" then, shall we?

**Comment by XPhyro** [comment]
> Is the bar the only place where font is used? If so then I can quickly raise a PR.

As far as I know, it is.

**Comment by N-R-K** [comment]
> I see, so let's spin up the `gh-pages` "_server_" then, shall we?

I dont mind.



>  If so then I can quickly raise a PR.

On second thought, lets merge this PR first otherwise we'll need to rebase everything. We can change `.font` to `.bar.font` after this has been merged. And once again, since we're merging to `dev` it should be fine.

**Comment by XPhyro** [comment]
Changed two things:
- Since this is replacing the thumbnail image, made the thumbnail window biggest.
- Made the title and bar bigger.



I'll decrease the resolution before this goes up.

I could probably screenshot a 16:9 image if you don't like 21:9.

-----

By the way, since this is on another branch, this PR can be merged without any problems. We could change the previews later.

**Comment by N-R-K** [comment]
16:9 would be good. One last thing, can you make the thumnail sizes bigger too?

- - -

Also moving to `src/`, it should be done on it's own PR as well. I've updated the [todo list](https://github.com/nsxiv/nsxiv/issues/24)

**Comment by eylles** [comment]
so i haven't got this to compile but i'm sure someone else will pick it up, here's a script to generate a new data.h without uncompressed image data
gendata.sh
```sh
#!/bin/sh
if [ "$1" = "undo" ] ; then
	[ -f old_data.h ] && mv old_data.h data.h
else
	[ -f old_data.h ] || mv data.h old_data.h
	head -n 9 old_data.h > data.h
	for i in 16x16.png 32x32.png 48x48.png 64x64.png 128x128.png ; do
		convert "$i" "${i%.png}.rgb"
		xxd -i "${i%.png}.rgb" >> data.h
		echo "\n# ${i} \n" >> data.h
		rm  "${i%.png}.rgb"
	done
	tail -n 13 old_data.h  >> data.h
fi
```
and the suggested change to the code
```diff
-		icon_data[n++] = icon_colors[icons[i].data[j] & 0x0F];
+		icon_data[n++] = icons[i].data[j];
```

mind you this will error out on compilation but i guess at least it gives a start for how to change the embedded icon.

**Comment by XPhyro** [comment]
```diff
		xxd -i "${i%.png}.rgb" >> data.h
-		echo "\n# ${i} \n" >> data.h
		rm  "${i%.png}.rgb"
	done
	tail -n 13 old_data.h  >> data.h
+       sed -i 's/__\([0-9]*\)x[0-9]*_rgb/icon_data_\1/' data.h
```

It compiles with this change, but segfaults.

**Comment by TAAPArthur** [comment]
I think we should wait to release a new version until https://github.com/nsxiv/nsxiv/pull/49. If we accept it then users would only have to update their configs once when moving to nsxiv and never be forced to do so again.

**Comment by N-R-K** [comment]
@TAAPArthur whatever we do, this PR gets merged first. Rebasing other PRs will be much easier than having to rebase this one.

**Comment by eylles** [comment]
i screwed the icon generation 😅, speaking of, do we delay merging this until the new icons can be embedded? @XPhyro already managed to compile with embedding the new icons the problem now is that nsxiv segfaults with them...

**Comment by N-R-K** [comment]
> do we delay merging this until the new icons can be embedded?

This PR is already packed with too many things, and it's blocking other PRs. Let's keep this PR as is and make future changes in different PRs.

File: nsxiv.1
Comment (N-R-K) [inline nsxiv.1:483]:
```suggestion
https://github.com/nsxiv/nsxiv
```

File: README.md
Comment (eylles) [inline README.md:25]:
```suggestion
* Display image information in status bar
* Display image name/path in X title
```

File: CHANGELOG.md
Comment (eylles) [inline CHANGELOG.md:15]:
```suggestion
  * Configurable X window title with the `-T` flag (#23)
```
this maintains consistency with with the rest of the changelog as every time an option was added it was mentioned along it's flag.

Comment (GRFreire) [reply CHANGELOG.md]: Writing it like this seems like the only option to configure the X window title is with the `-T` flag, but `nsxiv` also support changing this thru `config.h`.

File: README.md
Comment (GRFreire) [inline README.md:3]: Out of curiosity, do you want to keep all these "alternative names" or to decide on one?

**Review by eylles — APPROVED** [review]

File: README.md
Comment (XPhyro) [reply README.md:3]: https://github.com/nsxiv/nsxiv/discussions/47#discussioncomment-1333816

**Review by GRFreire — APPROVED** [review]

**Comment by N-R-K** [comment]
Since there's a lot of changes in here, lets wait for someone else to also take a look. Maybe we missed something.

After that, I think we should be good to squash and merge this.

**Comment by TAAPArthur** [comment]
>@TAAPArthur whatever we do, this PR gets merged first. Rebasing other PRs will be much easier than having to rebase this one.

Let me rephrase, before we cut the release, I think we should reach a decision on all outstanding breaking changes.

**Review by TAAPArthur — APPROVED** [review]

**Comment by XPhyro** [comment]
There is a bug in this PR. nsxiv does not call `~/.config/nsxiv/exec/image-info` or `key-handler`, but rather calls `image-inf` and `key-handle`.

**Comment by N-R-K** [comment]
Let's wait a bit more before merging this. Tomorrow I'll do some tests and re-review this with a clear head just to make sure there aren't any other bugs introduced.

**Review by kdkasad — APPROVED** [review]
I haven't done very extensive testing and I haven't looked at the changed code much, but I've been using it for a little while and everything seems to work as expected.

File: README.md
Comment (N-R-K) [inline README.md:20]:
```suggestion
* Customizable key and mouse button mappings (in *config.h*)
```

File: README.md
Comment (N-R-K) [inline README.md:32]:
```suggestion
**Image mode: (Default colors)**
```

File: README.md
Comment (N-R-K) [inline README.md:36]:
```suggestion
**Thumbnail mode: (Custom colors)**
```

**Review by N-R-K — CHANGES_REQUESTED** [review]

**Review by N-R-K — APPROVED** [review]

**Comment by N-R-K** [comment]
I think we're good. There shouldn't be any further bugs, and if any arise, we'll deal with it on the go.

**Pull request merged**
